Src/DependencyCollector/Microsoft.AI.DependencyCollector/DependencyCollectorDiagnosticListener.cs

```diff
@@ -0,0 +1,190 @@
+                telemetry.Data = requestUri.OriginalString;
+                this.pendingTelemetry.TryAdd(loggingRequestId, telemetry);
+
+                if (!request.Headers.Contains(RequestResponseHeaders.RequestContextSourceKey))
+                {
+                    string sourceApplicationId;
+                    if (this.correlationIdLookupHelper.TryGetXComponentCorrelationId(telemetry.Context.InstrumentationKey, out sourceApplicationId))
+                    {
+                        request.Headers.Add(RequestResponseHeaders.RequestContextSourceKey, sourceApplicationId);
+                    }
+                }
+
+                // Add the root ID
+                string rootId = telemetry.Context.Operation.Id;
+                if (!string.IsNullOrEmpty(rootId) && !request.Headers.Contains(RequestResponseHeaders.StandardRootIdHeader))
+                {
+                    request.Headers.Add(RequestResponseHeaders.StandardRootIdHeader, rootId);
+                }
+
+                // Add the parent ID
+                string parentId = telemetry.Id;
+                if (!string.IsNullOrEmpty(parentId) && !request.Headers.Contains(RequestResponseHeaders.StandardParentIdHeader))
+                {
+                    request.Headers.Add(RequestResponseHeaders.StandardParentIdHeader, parentId);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Diagnostic event handler method for 'System.Net.Http.Response' event. Even in the case of an exception, this will still be called.
+        /// See https://github.com/dotnet/corefx/blob/master/src/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs#L71 for more information.
+        /// </summary>
+        [DiagnosticName("System.Net.Http.Response")]
+        public void OnResponse(HttpResponseMessage response, Guid loggingRequestId)
+        {
+            if (response != null)
+            {
+                DependencyTelemetry telemetry;
+                if (this.pendingTelemetry.TryRemove(loggingRequestId, out telemetry))
+                {
+                    if (response.Headers.Contains(RequestResponseHeaders.RequestContextTargetKey))
+                    {
+                        string targetApplicationId = response.Headers.GetValues(RequestResponseHeaders.RequestContextTargetKey).SingleOrDefault();
+                        if (!string.IsNullOrEmpty(targetApplicationId))
+                        {
+                            // We only add the cross component correlation key if the key does not represent the current component.
+                            string sourceApplicationId;
+                            if (this.correlationIdLookupHelper.TryGetXComponentCorrelationId(telemetry.Context.InstrumentationKey, out sourceApplicationId) &&
+                                targetApplicationId != sourceApplicationId)
+                            {
+                                telemetry.Type = "Application Insights";
```

Comment:
"Application Insights" has changed to Http (tracked component) - see .net implemenation. I realize it's not part of your change but it's a simple string change in a single place - please go ahead and make the change.